shared/teams/invite-by-email/index.native.js

```diff
@@ -0,0 +1,230 @@
+    </Box>
+  </Box>
+)
+
+const contactRow = (i: number, props: ContactRowProps) => {
+  const source = typeof props.contact.thumbnailPath === 'string'
+    ? {uri: 'file://' + props.contact.thumbnailPath}
+    : props.contact.thumbnailPath
+  const hasThumbnail = props.contact.thumbnailPath && props.contact.thumbnailPath.length > 0
+  return (
+    <Box
+      style={{
+        ...globalStyles.flexBoxRow,
+        alignItems: 'center',
+        height: 56,
+        width: '100%',
+        padding: globalMargins.small,
+      }}
+    >
+      <Box style={{...globalStyles.flexBoxRow, alignItems: 'center', flex: 1}}>
+        <Box style={{...globalStyles.flexBoxRow, alignItems: 'center', flex: 1}}>
+          {!!hasThumbnail &&
+            <NativeImage
+              style={{width: 48, height: 48, borderRadius: 24, marginRight: 16}}
+              source={source}
+            />}
+          {!hasThumbnail && <Avatar size={48} style={{marginRight: 16}} />}
+          <Box>
+            <Box style={globalStyles.flexBoxRow}>
+              <Text type="BodySemibold">{props.contact.name}</Text>
+            </Box>
+            <Box style={globalStyles.flexBoxRow}>
+              <Text type="BodySmall">
+                {props.contact.email || props.contact.phoneNo}
+              </Text>
+            </Box>
+          </Box>
+        </Box>
+        <Box>
+          <Button
+            type={props.selected ? 'Following' : 'Primary'}
+            label={props.selected ? 'Invited!' : 'Invite'}
+            waiting={props.loading}
+            small={true}
+            onClick={props.onClick}
+            style={{width: 100, paddingLeft: globalMargins.small, paddingRight: globalMargins.small}}
+          />
+        </Box>
+      </Box>
+    </Box>
+  )
+}
+
+type State = {
+  filter: string,
+}
+
+class InviteByEmail extends React.Component<MobileProps, State> {
+  constructor(props: MobileProps) {
+    super(props)
+    this.state = {
+      filter: '',
+    }
+  }
+
+  _onSelectContact(contact: ContactDisplayProps) {
+    if (this._isSelected(contact.recordID)) {
+      this._removeInvitee(contact)
+    } else {
+      this._addInvitee(contact)
+    }
+  }
+
+  _isSelected(id: string): boolean {
+    return this.props.invited.findIndex(rec => rec.contactID === id) >= 0
+  }
+
+  _addInvitee(contact: ContactDisplayProps) {
+    if (contact.email) {
+      this.props.onInviteEmail(contact.email)
+    } else if (contact.phoneNo) {
+      this.props.onInvitePhone(contact.phoneNo)
+    }
+  }
+
+  _removeInvitee(contact: ContactDisplayProps) {
+    if (contact.email) {
+      this.props.onUninvite(contact.email)
+    }
+  }
+
+  _trim(s: ?string): string {
+    return (s && s.replace(/^[^a-z0-9@.]/i, '').toLowerCase()) || ''
+  }
+
+  render() {
+    // Filter before adding props to avoid a long map fcn
+    const filteredContactRows = this.props.contactRowProps.filter(contact => {
```

Comment:
i don't understand why we have complicated stuff here and also in the container, can't this move into the container or vice versa